activerecord/CHANGELOG.md

```diff
@@ -1,3 +1,25 @@
+*   Add option to merge association scopes to subqueries.
+
+    It removes a breaking change. Conditions that were not applied in
+    previous versions would be applied now giving people different results.
+
+    Set the option on an association:
+
+    ```ruby
+    class Author < ApplicationRecord
+      has_many :welcome_posts, -> { where(title: "welcome") } , class_name: "Post", apply_on_subqueries: true
+    end
+    ```
+
+    Then the association scope will be merged into the subquery.
+
+    ```ruby
+    Author.where(welcome_posts: Post.all)
```

Comment:
#48487 also broke on associations with a parameterized scope (`Author.where(posts_with_signature: Post.all)`). Now that it's optional that may be less of a concern, but we should avoid having accidentally-incompatible features... I don't _think_ this is inherently incompatible, so we should probably fix that.